control-plane-operator/controllers/hostedcontrolplane/configoperator/params.go

```diff
@@ -42,6 +43,38 @@ func NewHostedClusterConfigOperatorParams(ctx context.Context, hcp *hyperv1.Host
 			},
 		},
 	}
+	params.LivenessProbes = config.LivenessProbes{
+		hccContainerMain().Name: {
+			Handler: corev1.Handler{
+				HTTPGet: &corev1.HTTPGetAction{
+					Path:   "/metrics",
```

Comment:
why is this using /metrics for probes?

Reply:
Checking metrics endpoint is a good liveness test for this component since non-200 response from the endpoint clearly indicates pod health problem.

Reply:
Please enable the health endpoint instead and use that for probes rather than abusing the metrics endpoint: https://github.com/kubernetes-sigs/controller-runtime/blob/4d10a0615b11507451ecb58bfd59f0f6ef313a29/pkg/manager/manager.go#L204

Reply:
> 

@alvaroaleman code changed, please check if new implementation meets your expectations.